summary_workflows/quantification/match_with_gt.py

```diff
@@ -0,0 +1,187 @@
+import subprocess
+from io import StringIO
+import pandas as pd
+import numpy as np
+from shutil import which
+import argparse
+
+parser = argparse.ArgumentParser()
+
+parser.add_argument('a', help='The BED file containing predictions. MUST be BED6 format.')
+parser.add_argument('b', help='The ground truth bed file. First 6 columns must be standard BED6, but can have additional columns appended.')
+parser.add_argument('window', help='Number of bases to append to each side of the predicted site.', type=int)
+#parser.add_argument('-o', help='output file directory') # not yet implemented!!
+args = parser.parse_args()
+
+
+f_GT = args.b
+f_PD = args.a
+window = args.window
+
+
+def bedtools_window(bed1, bed2, window):
+    """
+    Python wrapper for bedtools window.
+    """
+    
+    # make sure bedtools can be called in current env
+    assert which('bedtools') is not None, "bedtools not installed or not in PATH"
+
+    # run bedtools window, capture output
+    out = subprocess.run(['bedtools', 'window', '-sm',
+                          '-w', str(window), 
+                          '-a', bed1, 
+                          '-b', bed2], 
+                         capture_output=True, shell=False)
+
+    assert out.returncode==0, "bedtools window run failed, check input files"
+
+    # memory file-handle to pass output to pandas without writing to disk
+    out_handle = StringIO(out.stdout.decode())
+    out = pd.read_csv(out_handle, delimiter='\t', header=None, dtype={0: str})
+    
+    return(out)
+
+def overlap(coords1, coords2):
+    """
+    Calculates number of points overlapping when the input are two intervals as tuples.
+    Input must be coords1=(int, int),  coords2=(int, int).
+    """
+    return(len(np.intersect1d(np.arange(coords1[0], coords1[1]), np.arange(coords2[0], coords2[1]))))
+
+
+print('-------------------------------------------------')
+## sum expression levels for predicted sites that overlap with each other given the window size
+## assuming the prediction bed file has exactly 6 columns in the correct order, no header!
+
+# find sites that overlap within the predict set itself when the window is extended
+out = bedtools_window(f_PD, f_PD, window)
+
+# for logging
+len_dup = len(out)
+
+not_unq_mask = out.duplicated([0, 1, 2, 5], keep=False)
```

Comment:
Having numbered columns makes it difficult to read the code. Would have been better if the columns had meaningful names.